File: java/BarangDagangan.java
// Kelas anak dari Senjata: menambah atribut dagang
public class BarangDagangan extends Senjata {

    private double  harga;   // Harga dalam Rupiah
    private int     stok;    // Jumlah stok
    private String  kondisi; // Baru / Bekas
    private boolean garansi; // True = Ya, False = Tidak

    // Constructor kosong
    public BarangDagangan() {
        super();
    }

    // Constructor dengan Parameter
    public BarangDagangan(String idSenjata, String nama, String jenis, String spesifikasi, String madeIn, double harga, int stok, String kondisi, boolean garansi) {
        super(idSenjata, nama, jenis, spesifikasi, madeIn);
        this.harga   = harga;
        this.stok    = stok;
        this.kondisi = kondisi;
        this.garansi = garansi;
    }

    // Getter Method
    public double  getHarga()   { return harga; }
    public int     getStok()    { return stok; }
    public String  getKondisi() { return kondisi; }
    public boolean getGaransi() { return garansi; }

    // Setter Method
    public void setHarga(double harga)     { this.harga = harga; }
    public void setStok(int stok)          { this.stok = stok; }
    public void setKondisi(String kondisi) { this.kondisi = kondisi; }
    public void setGaransi(boolean garansi) { this.garansi = garansi; }
}

File: cpp/BarangDagangan.cpp
#include "Senjata.cpp"

// Kelas anak dari Senjata: menambah atribut dagang (harga, stok, kondisi, garansi)
class BarangDagangan : public Senjata {

private:
    double harga;   // Harga dalam Rupiah
    int    stok;    // Jumlah stok
    string kondisi; // Baru / Bekas
    bool   garansi; // True = Ya, False = Tidak

public:
    // Constructor kosong
    BarangDagangan() {}

    // Constructor dengan Parameter
    BarangDagangan(string id, string n, string j, string s, string m, double h, int st, string k, bool g) {
        setIdSenjata(id);
        setNama(n);
        setJenis(j);
        setSpesifikasi(s);
        setMadeIn(m);
        harga   = h;
        stok    = st;
        kondisi = k;
        garansi = g;
    }

    // Getter Method
    double getHarga()const{
        return harga;
    }
    int getStok()const{
        return stok;
    }
    string getKondisi()const{
        return kondisi;
    }
    bool getGaransi()const{
        return garansi;
    }

    // Setter Method
    void setHarga(double h){
        harga = h;
    }
    void setStok(int st){
        stok = st;
    }
    void setKondisi(string k){
        kondisi = k;
    }
    void setGaransi(bool g){
        garansi = g;
    }

    // Destructor
    virtual ~BarangDagangan() {}
};
